File: CivitasJava/civitas/SorpresaPorCasaHotel.java
package civitas;

import java.util.ArrayList;

public class SorpresaPorCasaHotel extends Sorpresa {
    private int valor;
    
    SorpresaPorCasaHotel(int valor, String tex){
        super(tex);
        this.valor = valor;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) throws Exception{
        if (jugadorCorrecto(actual, todos) ){
            informe(actual, todos);
            todos.get(actual).modificarSaldo( valor * todos.get(actual).cantidadCasasHoteles() );
        }
    }
}

File: CivitasRuby/sorpresa_por_casa_hotel.rb
require_relative 'sorpresa'

module Civitas
class SorpresaPorCasaHotel < Sorpresa
  
  def initialize(valor, texto)
    super(texto)
    @valor = valor
  end
  
  def aplicar_a_jugador(actual, todos)
    if jugador_correcto(actual, todos)
      informe(actual, todos)
      todos[actual].modificar_saldo( @valor * todos[actual].cantidad_casas_hoteles)
    end
  end
  
  public_class_method :new
end
end
